Prevent inventory from flooding the log with gohai warnings

hush-hush opened the pull request:
### What does this PR do?

Prevent inventory from flooding the log with gohai warnings.

The PR also cleanup the many mutexes we have in the `inventories`.

We don't need control on data access in inventory. This commit replaces all the different mutexes we have in inventory by a single one. This should not impact performances as metadata is rarely added or updated. We now have a single mutex we lock when entering the module.

### Describe how to test/QA your changes

Start the agent on a host where gohai produce warnings. For example a linux host without `python` in PATH. The warning should only be log once at the start (and the every ~2h) instead of every 10min.

hush-hush committed:
Prevent inventory from flooding the log with gohai warnings

djmitche commented on pkg/metadata/inventories/inventories.go:
```diff
@@ -203,6 +210,25 @@ func createPayload(ctx context.Context, hostname string, ac AutoConfigInterface,
+	// every 1 to 10 min (depending on new metadata being registered). We don't want to log the same error again and again.
+	// We log once every 12 times on normal log level and on debug the rest of the time. The metadata in this paylaod should
+	// not change often, so with 1 paylaod every 10 minutes we would log once every 2h.
+	logLevelMutex.Lock()
+	defer logLevelMutex.Unlock()
```
What is this protecting?  It looks like, just `logCount`, since logErrorf and logInfof are read without holding this lock.  Assuming there's concurrency here (why else have mutexes?), that presents a race condition that might allow torn writes.

It might be better to just have an atomic to handle logCount, and access that within (fixed) wrapper functions logErrorf and logInfof.

hush-hush replied:
As we discussed I also cleanup the mutexes in the package to have a single one

hush-hush committed:
Clean mutexes in inventories

We don't need control on data access in inventory. This commit replaces
all the different mutexes we have in inventory by a single one. This
should not impact performances as metadata is rarely added or updated.

We now have a single mutex we lock when entering the module.
